lib/bamboo/adapters/rfc2822_with_bcc.ex

```diff
@@ -16,7 +16,8 @@ defmodule Bamboo.SesAdapter.RFC2822WithBcc do
   @address_types ["From", "To", "Reply-To", "Cc", "Bcc"]
 
   # https://tools.ietf.org/html/rfc2822#section-3.4.1
-  @email_validation_regex ~r/\w+@\w+\.\w+/
+  # https://gist.github.com/mgamini/4f3a8bc55bdcc96be2c6
+  @email_validation_regex ~r/^[A-Za-z0-9._%+-+']+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$/
```

Comment:
Let's use this like in [original file](https://github.com/DockYard/elixir-mail/blob/bc27eba27ea0895a2fc12f5ac00de3e944a7fd8f/lib/mail/renderers/rfc_2822.ex#L25)

```
@email_validation_regex Application.get_env(
                            :mail,
                            :email_regex,
                            ~r/[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,64}/
                          )``

Reply:
Great, thanks.

I ammended the commit and added a test